src/StreamJsonRpc/JsonMessageFormatter.cs

```diff
@@ -135,11 +172,9 @@ public Version ProtocolVersion
         /// <summary>
         /// Gets the <see cref="Newtonsoft.Json.JsonSerializer"/> used when serializing and deserializing method arguments and return values.
         /// </summary>
-        public JsonSerializer JsonSerializer { get; } = new JsonSerializer()
-        {
-            NullValueHandling = NullValueHandling.Ignore,
-            ConstructorHandling = ConstructorHandling.AllowNonPublicDefaultConstructor,
-        };
+        public JsonSerializer JsonSerializer { get; }
+
+        public JsonRpc Rpc { get; set; }
 
         /// <inheritdoc/>
```

Comment:
The getter can be `private` here.
But actually I think we should custom implement this property setter, as IMO it should throw if it is set to a non-null value when it is already non-null, as that would indicate a formatter is being used for multiple JsonRpc instances, which at this point would likely be very bad.

Reply:
Better, but not quite what I had in mind. This would make `value==null` only allowed when `this.rpc != null`. If we allow setting to null anytime, we ought to allow it being set to null all the time.
What I was suggesting was this: `Verify.Operation(this.rpc == null, "This formatter already belongs to another JsonRpc instance. Create a new instance of this formatter for each new JsonRpc instance.");`
And of course move the string into the resx file.

---
In reply to: [303662153](https://github.com/milopezc/vs-streamjsonrpc/pull/1#discussion_r303662153) [](ancestors = 303662153)